File: Multi_bracket_validation/multi-bracket-validation.js
'use strict';

let  multiBracketValidation = (string) => {

  let correctBracket = '[]{}()',
    stack = [],
    character;

  for(let i=0; character = string[i]; i++){
    let bracePosition = correctBracket.indexOf(character);
    if(bracePosition === -1) {
      continue;
    }else if(bracePosition % 2 === 0) {
      stack.push(bracePosition + 1);
    } else {
      if(stack.pop() !== bracePosition) {
        return false;
      }
    }
  }
  return stack.length === 0;

}

export default multiBracketValidation;

File: java401challenges/src/main/java/java401challenges/stacksandqueues/MultiBracketValidation.java
package java401challenges.stacksandqueues;
import java.util.Stack;

public class MultiBracketValidation {

    static Stack<Character> stack = new Stack<>();


    public static boolean multiBracketValidation(String input) {
        try {
            int i = 0;
            while(i < input.length()) {

                if(input.charAt(i) == '}' || input.charAt(i) == ')' || input.charAt(i) == ']' ){
                    stack.pop();
                }else if(input.charAt(i) == '{' || input.charAt(i) == '(' || input.charAt(i) == '[') {
                    stack.push(input.charAt(i));
                }
                i++;
            }
            if(stack.isEmpty()) return true;
            else return false;
        } catch (Exception e) {
            return false;
        }
    }
}
